### ISSUE TITLE
GPT2 Example job issues when supplying target_modules param in LoraConfig

### ISSUE THREAD

**Issue by cooleo-dude** [FederatedAI/FATE-LLM#38]
When following the GPT2 example, the following errors occur:
![image](https://github.com/FederatedAI/FATE-LLM/assets/79670678/10f54484-32f8-4985-8ac4-daae21afb7df)
After debugging, this error is caused by the json/encoder.py file when trying to reference the target_modules = ['c_attn'] component of the LoraConfig object. Changing this value to a string 'c_attn' resolves the issue, however limits the ability to fine-tune multiple categories. However, after doing so the following error occurs.
![image](https://github.com/FederatedAI/FATE-LLM/assets/79670678/3b1dfd3d-7004-4c07-b80f-ffe5eba1804b)
This occurs because the t.nn.CustModel object created in the pipeline job does not reference any of the layers from the GPT2 model, and there does not appear a way to do so. Is there a workaround for this, or would this be an environment issue?

**Comment by strongman1995** [comment]
I encounter the same issue as well @mgqa34

**Comment by cooleo-dude** [comment]
Solved the issue. default function inside /env/python/miniconda/lib/python3.8/json/encoder.py is not implemented and causes issues when serializing. Creating this function removes the issues and job runs as intended. Check line 161 of this file and make changes accordingly. I have attached a screenshot below to show the changes I made to make it work.

![image](https://github.com/FederatedAI/FATE-LLM/assets/79670678/8d6e0357-79d5-497c-82b8-dcd40900420d)
